src/python/pants/engine/target.py

```diff
@@ -606,6 +606,22 @@ def expect_single(self) -> Target:
         return self[0]
 
 
+@dataclass(frozen=True)
+class CoarsenedTarget:
+    """A set of Targets which cyclicly reach one another, and are thus indivisable."""
+
+    # The members of the cycle.
+    members: Tuple[Target, ...]
+    # The deduped dependencies of all Targets in the cycle. Dependencies between members of the
+    # cycle are excluded, so this set of dependencies will only refer to other/disjoint
+    # CoarsenedTarget instances.
+    dependencies: FrozenOrderedSet[Address]
```

Comment:
This comment confuses me as I would expect dependencies to be `FrozenOrderedSet[CoarsenedTarget]`. Perhaps this?

> The deduped direct (not transitive) dependencies of all Targets in the cycle. Dependencies between members of the cycle are excluded.
> 
> To expand these dependencies, e.g. to get the transitive dependencies, use `await Get(...)`.

The second paragraph is meant to explain how you might consume `dependencies` - I'm not following it rn.